src/python/pants/engine/internals/graph.py

```diff
@@ -228,7 +200,7 @@ async def resolve_target_parametrizations(
         and target_type.deprecated_alias == target_adaptor.type_alias
         and not address.is_generated_target
     ):
-        await Get(_WarnDeprecatedTarget, _WarnDeprecatedTargetRequest(target_type))
+        warn_deprecated_target_type(target_type)
```

Comment:
These can both move into the `Target` `__init__` method instead I think.

Reply:
I don't have the `target_adaptor.type_alias` in the constructor. But I can move the field type deprecations check.

Reply:
Ran into an issue when moving to the `Target` constructor, so will leave it as TODO for now.